src/solvers/slvr_blk_2m.hpp

```diff
@@ -0,0 +1,399 @@
+#pragma once
+#include "slvr_sgs.hpp"
+
+#include <libcloudph++/blk_2m/options.hpp>
+#include <libcloudph++/blk_2m/rhs_cellwise.hpp>
+#include <libcloudph++/blk_2m/rhs_columnwise.hpp>
+
+
+template <class ct_params_t>
+class slvr_blk_2m_common : public std::conditional_t<ct_params_t::sgs_scheme == libmpdataxx::solvers::iles,
```

Comment:
Plenty of similarities to the slvr_blk_1m_common class.
Would be nice to make a slrv_blk_common class from which blk_1m and blk_2m would inherit.

Reply:
Yes, @trontrytel and I agree. That was the idea of the `common_water_src` function (which is still commented out). But as a first step I hadn't worried about refactoring yet.